guile: update to version 2.2.0

ecbrown opened the pull request:
Update guile to version 2.2.0

Trac ticket: https://trac.macports.org/ticket/53830

Tested on

macOS 10.12.3
Xcode 8.2.1

Verification

Have you

- [x] followed our Commit Message Guidelines?
- [x] checked that there aren't other open pull requests for the same change?
- [x] referenced existing tickets on Trac with full URL?
Sorry, I have already opened the ticket, I will try to close it.

- [x] checked your Portfile with port lint?
- [x] tried existing tests with sudo port test? (no tests turned on)
- [x] tried a full install with sudo port -vst install?
- [x] tested basic functionality of all binary files?

ecbrown commented:
Guile has a number of dependent ports, and each seems to compile and run, except for gnotime (which seems to want guile 2.0).

mojca commented on lang/guile/Portfile:
```diff
@@ -3,7 +3,8 @@
 name                guile
-version             2.0.14
+version             2.2.0
+revision            1
```
Why would you set the revision to 1?

ecbrown replied:
This may be an error.  I recall, for another port, to add a revision number.  Perhaps it is zero-indexed, and if the first (0th) revision, then this is not necessary.

mojca commented on lang/guile/Portfile:
```diff
@@ -58,9 +59,6 @@ configure.args      CPPFLAGS="-I${prefix}/include" \
                     --disable-error-on-warning \
                     --disable-silent-rules
 
-patchfiles          sierra.patch \
```
If you removed patchfiles (probably because they were included upstream already?), you could probably remove the files themselves as well.

ecbrown replied:
my mistake.  I had removed it in my first attempt, but forgot it this time.

ecbrown commented:
I have made a few changes (which appear as commits on this merge request)

* removed revision number line
* removed patch files

(I don't think the guile dependencies need to be altered in any way.)

ecbrown commented:
Looks like this commit will break gmake +guile, so I think I have some more work to do before this is ready for prime time...

Schamschula commented:
In the meantime the guile Portfile has been updated to version 2.2.2.
